src/mbgl/style/expression/in.cpp

```diff
@@ -0,0 +1,116 @@
+#include <string.h>
+#include <mbgl/style/conversion_impl.hpp>
+#include <mbgl/style/expression/in.hpp>
+#include <mbgl/style/expression/type.hpp>
+#include <mbgl/util/string.hpp>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+EvaluationResult In::evaluate(const EvaluationContext& params) const {
+    const EvaluationResult evaluatedNeedle = needle->evaluate(params);
+    const EvaluationResult evaluatedHeystack = haystack->evaluate(params);
+
+    if (!evaluatedNeedle) {
+        return evaluatedNeedle.error();
+    }
+    if (!evaluatedHeystack) {
+        return evaluatedHeystack.error();
+    }
+
+    type::Type evaluatedNeedleType = typeOf(*evaluatedNeedle);
+    if (!isComparableRuntimeValue(evaluatedNeedleType)) {
+        return EvaluationError{"Expected first argument to be of type boolean, string or number, but found " +
+                               toString(evaluatedNeedleType) + " instead."};
+    }
+
+    type::Type evaluatedHeystackType = typeOf(*evaluatedHeystack);
+    if (!isSearchableRuntimeValue(evaluatedHeystackType)) {
+        return EvaluationError{"Expected second argument to be of type array or string, but found " +
+                               toString(evaluatedHeystackType) + " instead."};
+    }
+
+    if (evaluatedNeedleType == type::Null || evaluatedHeystackType == type::Null) {
+        return EvaluationResult(false);
+    }
+
+    if (evaluatedHeystackType == type::String) {
+        const auto heystackString = evaluatedHeystack->get<std::string>();
+        std::string needleValue = "";
+        if (evaluatedNeedleType == type::Boolean) {
+            needleValue = evaluatedNeedle->get<bool>() ? "true" : " false";
+        } else if (evaluatedNeedleType == type::String) {
+            needleValue = evaluatedNeedle->get<std::string>();
+        } else if (evaluatedNeedleType == type::Number) {
+            needleValue = std::to_string(evaluatedNeedle->get<double>());
+            needleValue.erase(needleValue.find_last_not_of('0') + 1, std::string::npos);
+            needleValue.erase(needleValue.find_last_not_of('.') + 1, std::string::npos);
```

Comment:
Could you explain why this is required.

Reply:
Since we use `evaluatedNeedle->get<double>()` to get a double value, it will has 6 tail 0. For example, if needle is 6, `needleValue` will be "6.000000", which will cause find failed.